src/outputs/parthenon_hdf5.cpp

```diff
@@ -415,17 +488,26 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
     ~MPI_InfoDeleter() { MPI_Info_free(&info); }
   } delete_info{FILE_INFO_TEMPLATE};
 
-  PARTHENON_HDF5_CHECK(H5Pset_sieve_buf_size(acc_file, 262144));
-  PARTHENON_HDF5_CHECK(H5Pset_alignment(acc_file, 524288, 262144));
-
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "access_style", "write_once"));
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "collective_buffering", "true"));
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_block_size", "1048576"));
-  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_buffer_size", "4194304"));
+  // Hint specifies the manner in which the file will be accessed until the file is closed
+  char* access_style = HDF5::get<char *>("access_style", (char*)"write_once", exists);
+  PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "access_style", access_style));
+
+  // Specifies whether the application may benefit from collective buffering
+  // Default for collective_buffering is disabled
+  bool collective_buffering = HDF5::get<bool>("collective_buffering", false, exists);
+  if(collective_buffering) {
+    PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "collective_buffering", "true"));
+    // Specifies the block size to be used for collective buffering file acces
+     char *cb_block_size = HDF5::get<char *>("cb_block_size", (char*)"1048576", exists);
+    PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_block_size",  cb_block_size));
+    // Specifies the total buffer space that can be used for collective buffering on each target node, 
+    // usually a multiple of cb_block_size
+    char *cb_buffer_size = HDF5::get<char *>("cb_buffer_size", (char*)"4194304", exists);
+    PARTHENON_MPI_CHECK(MPI_Info_set(FILE_INFO_TEMPLATE, "cb_buffer_size", cb_buffer_size));
+  }
 
   /* tell the HDF5 library that we want to use MPI-IO to do the writing */
   PARTHENON_HDF5_CHECK(H5Pset_fapl_mpio(acc_file, MPI_COMM_WORLD, FILE_INFO_TEMPLATE));
-  PARTHENON_HDF5_CHECK(H5Pset_fapl_mpio(acc_file, MPI_COMM_WORLD, MPI_INFO_NULL));
```

Comment:
It does not look like collective_buffering was ever enabled, the second call to the same function would override the MPI info with MPI_INFO_NULL.